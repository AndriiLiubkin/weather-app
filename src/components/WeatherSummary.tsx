import React from 'react';
import moment from 'moment';
import { IWeatherResponseDTO } from "../types";

const WeatherSummary = ({ weatherData}: { weatherData: IWeatherResponseDTO[] }) => {
  
  const findBestUmbrella = (weatherData: IWeatherResponseDTO[]) => {
    const result = weatherData.map((d: IWeatherResponseDTO) => d.weather[0].main === "Rain").lastIndexOf(true);
    if(result === -1) {
      return weatherData[0].dt_txt;
    }
    return weatherData[result].dt_txt;
  }

  const findBestJaket = (weatherData: IWeatherResponseDTO[]) => {
    const result = weatherData.reduce((prev: IWeatherResponseDTO, curr: IWeatherResponseDTO) => {
      return prev.main.temp < curr.main.temp ? prev : curr;
    });

    return result.dt_txt;
  }
  
  return (
    <div className="weather-summary">
      <h2>Weather Summary:</h2>
      {
        weatherData.length > 0 && 
          <ul className="summary-info">
            <li>
              <h3>Best day to sell an umbrella:</h3>
              <p>After {moment(findBestUmbrella(weatherData)).format('dddd')}</p>
            </li>
            <li>
              <h3>Best day to sell jacket:</h3>
              <p>After {moment(findBestJaket(weatherData)).format('dddd')}</p>
            </li>
          </ul>
      }
    </div>
  )
}

export default WeatherSummary;
