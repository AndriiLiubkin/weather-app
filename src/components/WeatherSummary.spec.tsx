import WeatherSummary from "./WeatherSummary";

import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

const mockWeatherListData = [
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
  {
    dt: 1632495600,
    dt_txt: "2021-09-24 15:00:00",
    main: {
      feels_like: 22.17,
      humidity: 50,
      pressure: 1012,
      temp: 22.55,
    },
    weather: [
      {
        description: "clear sky",
        icon: "01d",
        id: 800,
        main: "Rain",
      },
    ],
    pop: 53,
    visibility: 21,
    wind: {
      deg: 279,
      gust: 7.07,
      speed: 3.92,
    },
  },
];

Enzyme.configure({ adapter: new Adapter() });

describe("<WeatherSummary />", () => {
  it("test empty weather summary", () => {
    const wrapper = shallow(<WeatherSummary weatherData={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("test weather summary", () => {
    const wrapper = shallow(
      <WeatherSummary weatherData={mockWeatherListData} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
