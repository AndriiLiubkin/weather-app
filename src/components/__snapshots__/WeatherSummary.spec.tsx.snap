// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherSummary /> test empty weather summary 1`] = `ShallowWrapper {}`;

exports[`<WeatherSummary /> test weather summary 1`] = `ShallowWrapper {}`;

exports[`<WeatherSummary /> test weather summary 2`] = `ShallowWrapper {}`;
